pallets/staking-rewards/src/mock.rs

```diff
@@ -701,3 +760,209 @@ pub(crate) fn chain_state() -> (u128, u128, u128, u128) {
         StakingRewards::pool(),
     )
 }
+
+pub(crate) mod two_block_producers {
+    use super::*;
+
+    pub(crate) type SignedExtra = pallet_gear_staking_rewards::StakingBlackList<Test>;
+    type TestXt = sp_runtime::testing::TestXt<RuntimeCall, SignedExtra>;
+    type Block = TestBlock<TestXt>;
+
+    construct_runtime!(
+        pub enum Test
+        {
+            System: system,
+            Timestamp: pallet_timestamp,
+            Authorship: pallet_authorship,
+            Balances: pallet_balances,
+            Staking: pallet_staking,
+            Session: pallet_session,
+            Historical: pallet_session_historical,
+            Treasury: pallet_treasury,
+            BagsList: pallet_bags_list::<Instance1>,
+            Sudo: pallet_sudo,
+            Utility: pallet_utility,
+            ElectionProviderMultiPhase: multi_phase,
+            StakingRewards: pallet_gear_staking_rewards,
+        }
+    );
+
+    impl<C> frame_system::offchain::SendTransactionTypes<C> for Test
+    where
+        RuntimeCall: From<C>,
+    {
+        type OverarchingCall = RuntimeCall;
+        type Extrinsic = TestXt;
+    }
+
+    common::impl_pallet_system!(Test, DbWeight = RocksDbWeight, BlockWeights = ());
+    common::impl_pallet_timestamp!(Test);
+
+    pub struct BlockAuthor;
+
+    impl FindAuthor<AccountId> for BlockAuthor {
```

Comment:
can't we just use two producers for pallet-staking-rewards testing?
single-validator case covered by pallet-gear, I guess